wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceDetailsViewActionsHandler.swift

```diff
@@ -125,6 +125,25 @@ final class DeviceDetailsViewActionsHandler: DeviceDetailsViewActions, Observabl
     func copyToClipboard(_ value: String) {
         UIPasteboard.general.string = value
     }
+
+    func downloadE2EIdentityCertificate() {
+        guard let certificate = certificate else {
+            return
+        }
+        let fileName = "e2eiCertifcate.txt"
+        let path = getDocumentsDirectory().appendingPathComponent(fileName)
+        do {
+            try certificate.certificateDetails.write(to: path, atomically: true, encoding: String.Encoding.utf8)
```

Comment:
where does it store it exactly ? does the user needs to see it?

We'll need to delete the file on logout

Reply:
Don't we delete all the files upon logout?

Reply:
This is supposed to invoke system to handle file saving. I will do that part of business logic implementation.